proto/internal.proto

```diff
@@ -121,17 +121,23 @@ enum PushTxnType {
 message InternalPushTxnRequest {
   optional RequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
   optional Transaction pushee_txn = 2 [(gogoproto.nullable) = false];
+  // Now optionally holds the timestamp used to compare the last heartbeat
```

Comment:
I found this comment hard to follow since it talks in terms of the low-level mechanism. How about:

Now is the time when this request was issued (captured in the request by the sender so all replicas have a consistent view of time). The current time is used instead of the transaction's timestamp in order to handle retries and detect abandoned transactions.